tests/ReplayData/WorkflowRun.test_rerun_failed_jobs.txt

```diff
@@ -0,0 +1,21 @@
+
+https
+POST
+api.github.com
+None
+/repos/PyGithub/PyGithub/actions/runs/9352357011/rerun-failed-jobs
+{'Authorization': 'Basic login_and_password_removed', 'User-Agent': 'PyGithub/Python'}
+None
+403
+[('Server', 'GitHub.com'), ('Date', 'Mon, 03 Jun 2024 15:10:59 GMT'), ('Content-Type', 'application/json; charset=utf-8'), ('Transfer-Encoding', 'chunked'), ('X-OAuth-Scopes', 'repo, workflow'), ('X-Accepted-OAuth-Scopes', ''), ('github-authentication-token-expiration', '2024-07-03 14:43:25 UTC'), ('X-GitHub-Media-Type', 'github.v3; format=json'), ('x-github-api-version-selected', '2022-11-28'), ('X-RateLimit-Limit', '5000'), ('X-RateLimit-Remaining', '4503'), ('X-RateLimit-Reset', '1717429911'), ('X-RateLimit-Used', '497'), ('X-RateLimit-Resource', 'core'), ('Access-Control-Expose-Headers', 'ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset'), ('Access-Control-Allow-Origin', '*'), ('Strict-Transport-Security', 'max-age=31536000; includeSubdomains; preload'), ('X-Frame-Options', 'deny'), ('X-Content-Type-Options', 'nosniff'), ('X-XSS-Protection', '0'), ('Referrer-Policy', 'origin-when-cross-origin, strict-origin-when-cross-origin'), ('Content-Security-Policy', "default-src 'none'"), ('Vary', 'Accept-Encoding, Accept, X-Requested-With'), ('Content-Encoding', 'gzip'), ('X-GitHub-Request-Id', '0402:1C44A8:1047CCA:1114381:665DDD03')]
+{"message":"Must have admin rights to Repository.","documentation_url":"https://docs.github.com/rest/actions/workflow-runs#re-run-failed-jobs-from-a-workflow-run"}
```

Comment:
Looks like your authentication does not allow for rerunning jobs. Please try to get a successful response here.

Reply:
Thanks @EnricoMi. I'm very happy to try to record a successful request. I copied the approach from the regular rerun method [which also has a failure](https://github.com/PyGithub/PyGithub/blob/main/tests/ReplayData/WorkflowRun.test_rerun.txt)

The problem I was having with getting a successful request is that the test references the PyGithub repository, which I don't have access to. I could attempt to mock up the recorded data by recording it against my own repository, and then doing a find-and-replace to make the repository name and workflow run ID match?

Reply:
I've pushed a change which should make the recording look mostly like what it should for a successful request. Hopefully that is okay.

I'm happy to change the test though, if you think it would be better to provide a fully authentic recording for a repository that I have admin access to though.